Cython/Compiler/ModuleNode.py

```diff
@@ -2318,12 +2327,22 @@ def generate_module_init_func(self, imported_modules, env, code):
         code.enter_cfunc_scope(self.scope)
         code.putln("")
         code.putln(UtilityCode.load_as_string("PyModInitFuncType", "ModuleSetupCode.c")[0])
-        header2 = "__Pyx_PyMODINIT_FUNC init%s(void)" % env.module_name
+        try:
+            env.module_name.encode('ascii')
+        except UnicodeEncodeError:
+            py2_mod_name = "PYTHON2_DOES_NOT_SUPPORT_UNICODE_MODULE_NAMES"
```

Comment:
I think the `#error` below is enough. This name wouldn't normally be user visible, I think. Just use the same name part as in Py3 here.

Reply:
OK. I've used a version with unicode characters dropped as the placeholder here instead